qiskit/algorithms/optimizers/de.py

```diff
@@ -0,0 +1,134 @@
+            popsize: Number of individuals/candidates in each generation
+            mutation: Mutation constant (differential weight) specified as a float in the range of
+                      [0,2] or as a tuple(min, max). If specified as a tuple, the mutation constant
+                      will be 'dithered' or randomly changed within the range specified in each
+                      generation.
+            recombination: Recombination constant (crossover probability) specified as a float
+                           in the range [0,1]
+            init: String/array to specify whether initial population is generately via ``random``,
+                  ``latinhypercube`` sampling or via a specified array of individuals. If an array
+                  is specified, its shape must match (popsize, params) where popsize and params
+                  refer to population size and number of parameters in the objective function
+            tol: Relative tolerance for convergence
+            atol: Absolute tolerance for convergence
+            polish: Boolean to specify whether or not best population in last generation is
+                    optimized using scipy.optimize.minimize with the L-BFGS-B method. If
+                    this is set to true, the maximum number of function evaluations will
+                    be greater than (maxiter + 1)* popsize * len(x)
+            disp: Boolean to specify if the objective function is printed at each iteration
+            workers: Integer or map-like callable (multiprocessing.Pool.map) that specifies
+                     how many individuals in the population to be evaluated in parallel
+        """
+
+        super().__init__()
+
+        self._strategy = strategy
+        self._maxiter = maxiter
+        self._popsize = popsize
+        self._mutation = mutation
+        self._recombination = recombination
+        self._tol = tol
+        self._atol = atol
+        self._init = init
+        self._polish = polish
+        self._disp = disp
+        self._workers = workers
+
+    def get_support_level(self) -> Dict[str, OptimizerSupportLevel]:
+        """ Returns support level dictionary. """
+        return {
+            'gradient': OptimizerSupportLevel.ignored,
+            'bounds': OptimizerSupportLevel.required,
+            'initial_point': OptimizerSupportLevel.ignored
+        }
+
+    def optimize(self,
+                 num_vars: int,
+                 objective_function: Callable,
+                 gradient_function: Callable = None,
+                 variable_bounds: List[Tuple[float, float]] = None,
+                 initial_point: np.ndarray = None) -> Tuple[np.ndarray, float, int]:
+        """ Runs the optimization. """
+        super().optimize(num_vars, objective_function, gradient_function,
+                         variable_bounds, initial_point)
+
+        res = differential_evolution(func=objective_function, bounds=variable_bounds, args=(),

```

Comment:
I note this function has a `seed` presumably to control random aspects. I am thinking we should set this to algorithm_globals.random_seed so that it runs from the common seed; I am not sure its worth exposing seed as a param in its own right as normally we control the randomness, so we have repeatability for test cases etc via utils/algorithm_globals module.

Reply:
Okay I can do this!